src/library/scala/collection/immutable/Vector.scala

```diff
@@ -888,35 +888,35 @@ private[immutable] trait VectorPointer[T] {
         display3 = copyOf(display3)
         display2 = copyOf(display2)
         display1 = copyOf(display1)
-        display5((index >> 25) & 31) = display4
-        display4((index >> 20) & 31) = display3
-        display3((index >> 15) & 31) = display2
-        display2((index >> 10) & 31) = display1
-        display1((index >>  5) & 31) = display0
+        display5((index >>> 25) & 31) = display4
```

Comment:
If there is error-checking to make sure they're positive or zero, then there is no reason to bother changing. If there is not error-checking, an exception could be replaced with a silently wrong result.  I think they should stay signed!